app/models/volunteer_op.rb

```diff
@@ -34,4 +42,16 @@ def self.contains_description?(text)
   def self.contains_title?(text)
     arel_table[:title].matches(text)
   end
+  
+  def address_complete?
+    address.present? && postcode.present?
+  end 
+  
+  private
+  
+  def clear_lat_lng
+    return if address_complete?
+    self.longitude = nil
+    self.latitude = nil
+  end
```

Comment:
@tansaku I already added a feature test in this PR https://github.com/AgileVentures/LocalSupport/pull/335, it makes sense to add the feature test in this PR because it allows to catch the problem in case.

And we can after refactor the model spec tests ( and add a unit test to check the conditional `after_validation` callback) and maybe we can add also an enhancement to check if imported volunteer op have coordinates before persisting them in the DB.